### PR TITLE
Add task to check if WordPress is installed otherwise install

### LINKED ISSUE #29: Install WordPress if not installed
Similar to the DB update check if `wp core is-installed` and based on `MULTISITE` in `.env` run `wp core install`
https://github.com/wearerequired/composer-deployer/blob/f288e34abbe3195b1395eee2ea505f7d6ef25cc5/res/deploy.tpl.php#L102-L122

### PR DESCRIPTION
**Pull request by grappler**
fixes #29

### CODE REVIEW

File: res/deploy.tpl.php [hunk 1]
```diff
@@ -34,6 +34,35 @@ function () {
 inventory( 'deploy.yml' );
 
 // Tasks.
+desc( 'Check WordPress is installed' );
+task(
+	'wp:maybe_installed',
```
Comment (ocean90) [inline res/deploy.tpl.php]:
```suggestion
	'wp:install',
```

`maybe_installed` sounds like it only checks but does not install. Let's just name it `wp:install`?

File: res/deploy.tpl.php [hunk 2]
```diff
@@ -34,6 +34,35 @@ function () {
...
+			function (): void {
+				$is_installed = test( '{{bin/wp}} core is-installed' );
+				if ( $is_installed ) {
+					return;
+				}
+				writeln( '<comment>Installing WordPress</comment>' );
+				$url          = run( 'set -o allexport; source wordpress/.env; set +o allexport; echo $_HTTP_HOST' );
+				$is_multisite = run( 'set -o allexport; source wordpress/.env; set +o allexport; echo $MULTISITE' );
+				if ( $is_multisite ) {
+					run( "{{bin/wp}} core multisite-install --url=https://{$url} --title='New Install' --admin_user=required --admin_email=info@required.ch --skip-email --skip-config" );
```
Comment (ocean90) [inline res/deploy.tpl.php]:
```suggestion
					run( "{{bin/wp}} core multisite-install --url=https://{$url} --title=WordPress --admin_user=required --admin_email=info@required.ch --skip-email --skip-config" );
```

Keep it simple.

File: res/deploy.tpl.php [hunk 3]
```diff
@@ -34,6 +34,35 @@ function () {
...
+				} else {
+
```
Comment (ocean90) [inline res/deploy.tpl.php]:
```suggestion
```

File: res/deploy.tpl.php [hunk 4]
```diff
@@ -34,6 +34,35 @@ function () {
...
+					run( "{{bin/wp}} core install --url=https://{$url} --title='New Install' --admin_user=required --admin_email=info@required.ch --skip-email" );
```
Comment (ocean90) [inline res/deploy.tpl.php]:
```suggestion
					run( "{{bin/wp}} core install --url=https://{$url} --title=WordPress --admin_user=required --admin_email=info@required.ch --skip-email" );
```

File: res/deploy.tpl.php [hunk 5]
```diff
@@ -34,6 +34,35 @@ function () {
...
+
+		within(
+			'{{release_path}}',
+			function (): void {
+				$is_installed = test( '{{bin/wp}} core is-installed' );
+				if ( $is_installed ) {
+					return;
+				}
+				writeln( '<comment>Installing WordPress</comment>' );
+				$url          = run( 'set -o allexport; source wordpress/.env; set +o allexport; echo $_HTTP_HOST' );
```
Comment (ocean90) [inline res/deploy.tpl.php]: Maybe name this `$host` or use `echo "https://$_HTTP_HOST"` to make it an URL?

**Review by ocean90 — CHANGES_REQUESTED** [review]

File: res/deploy.tpl.php (hunk 5 above)
Comment (grappler) [inline res/deploy.tpl.php]:
```suggestion
				$url          = run( 'set -o allexport; source wordpress/.env; set +o allexport; echo "https://$_HTTP_HOST" );
```

File: res/deploy.tpl.php (hunk 2 above)
Comment (grappler) [inline res/deploy.tpl.php]:
```suggestion
					run( "{{bin/wp}} core multisite-install --url={$url} --title=WordPress --admin_user=required --admin_email=info@required.ch --skip-email --skip-config" );
```

File: res/deploy.tpl.php (hunk 4 above)
Comment (grappler) [inline res/deploy.tpl.php]:
```suggestion
					run( "{{bin/wp}} core install --url={$url} --title=WordPress --admin_user=required --admin_email=info@required.ch --skip-email" );
```

File: res/deploy.tpl.php [hunk 6]
```diff
@@ -133,6 +161,7 @@ function (): void {
 		'deploy:shared',
 		'deploy:writable',
 		'deploy:vendors',
+		'wp:maybe_installed',
```
Comment (grappler) [inline res/deploy.tpl.php]:
```suggestion
		'wp:install',
```

File: res/deploy.tpl.php [hunk 7]
```diff
@@ -34,6 +34,34 @@ function () {
...
+		within(
+			'{{release_path}}',
+			function (): void {
+				$is_installed = test( '{{bin/wp}} core is-installed' );
+				if ( $is_installed ) {
+					return;
+				}
+				writeln( '<comment>Installing WordPress</comment>' );
+				$url          = run( 'set -o allexport; source wordpress/.env; set +o allexport; echo "https://$_HTTP_HOST"' );
+				$is_multisite = run( 'set -o allexport; source wordpress/.env; set +o allexport; echo $MULTISITE' );
```
Comment (ocean90) [inline res/deploy.tpl.php]: I think we can combine this into one command. Something like this should work:

```php
$env = run( 'set -o allexport; source wordpress/.env; set +o allexport; echo "https://$_HTTP_HOST,$MULTISITE"' );
[ $url, $is_multisite ] = explode( ',', $env );
```

**Comment by grappler** [comment]
I have changed the code and tested it again

**Review by ocean90 — APPROVED** [review]

**Pull request merged**